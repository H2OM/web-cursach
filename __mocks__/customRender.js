import ClientContext from '@/lib/context/ClientContext';
import { render } from '@testing-library/react';

export default customRender = (ui, {providerProps}) => {
    return render(
      <ClientContext.Provider value={{...providerProps}}>{ui}</ClientContext.Provider>,
    )
}