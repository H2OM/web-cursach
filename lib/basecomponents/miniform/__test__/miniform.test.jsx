import { screen } from "@testing-library/react";
import customRender from "@/__mocks__/customRender";
import userEvent from "@testing-library/user-event";
import settingFetchResult from "@/__mocks__/fetchMock";
import Miniform from "../miniform";
afterEach(() => {    
    jest.clearAllMocks();
  });

describe('test miniform', ()=>{
    
    it('test render', async ()=>{
        const {container} = customRender(<Miniform/>, {providerProps: {setNotification() {}}});

        expect(await screen.getByRole('button')).toBeInTheDocument();
        expect(await screen.getByPlaceholderText('Ваша почта')).toBeInTheDocument();
        expect(await screen.getByText('Неверный формат почты')).toBeInTheDocument();

        expect(container).toMatchSnapshot();
    });
    
    it('functional test success way', async ()=>{
        settingFetchResult({message: "success"});
        const spy = jest.spyOn(global, 'fetch');
        customRender(<Miniform/>, {providerProps: {setNotification() {}}});
        const input = screen.getByPlaceholderText('Ваша почта');
        const submit = screen.getByRole('button');

        await userEvent.type(input, 'someemail@gmail.ru');
        await userEvent.click(submit);

        expect(spy).toHaveBeenCalled();
        expect(input).toHaveValue("");
    });
    it('functional test exeption', async ()=>{
        const spy = jest.spyOn(global, 'fetch');
        customRender(<Miniform/>, {providerProps: {setNotification() {}}});
        const input = screen.getByPlaceholderText('Ваша почта');
        const submit = screen.getByRole('button');

        await userEvent.type(input, 'someemail@gmailru');
        await userEvent.click(submit);

        expect(spy).not.toHaveBeenCalled();
        expect(input).toHaveValue("someemail@gmailru");
    });
});
