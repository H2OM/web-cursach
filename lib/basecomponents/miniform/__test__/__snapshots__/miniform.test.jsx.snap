// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test miniform test render 1`] = `
<div>
  <form
    class="Miniform "
  >
    <div
      class="Miniform__input__wrap "
    >
      <input
        class="Miniform__input "
        name="mail"
        placeholder="Ваша почта"
        required=""
        type="text"
      />
      <label
        class="Miniform__input__error "
        for="email"
      >
        Неверный формат почты
      </label>
    </div>
    <button
      class="Miniform__button "
      type="submit"
    >
      Подписаться
    </button>
  </form>
</div>
`;
